### ISSUE TITLE
Gradle 5.0 warning in M4

### ISSUE THREAD

**Issue by jexp** [micronaut-projects/micronaut-core#487]
When creating and running an M4 app with gradle it issues this warning.

```
Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
See https://docs.gradle.org/4.8/userguide/command_line_interface.html#sec:command_line_warnings

A problem was found with the configuration of task ':shadowJar'. Registering invalid inputs and outputs via TaskInputs and TaskOutputs methods has been deprecated and is scheduled to be removed in Gradle 5.0.
 - No value has been specified for property 'mainClassName'.


> Task :compileTestGroovy
The following annotation processors were detected on the compile classpath: 'io.micronaut.annotation.processing.TypeElementVisitorProcessor' and 'io.micronaut.annotation.processing.PackageConfigurationInjectProcessor' and 'io.micronaut.annotation.processing.BeanDefinitionInjectProcessor'. Detecting annotation processors on the compile classpath is deprecated and Gradle 5.0 will ignore them. Please add them to the annotation processor path instead. If you did not intend to use annotation processors, you can use the '-proc:none' compiler argument to ignore them.
```

There might be more.

**Comment by jameskleeh** [comment]
@jexp What command was used to generate the app?

**Comment by jeffbrown** [comment]
Are you using your own local build of Gradle 5.0?

**Comment by jeffbrown** [comment]
Nevermind that question.  I misunderstood.  I thought you were using Gradle 5.0.  Apologies for the confusion.

**Comment by jeffbrown** [comment]
> What command was used to generate the app?

@jameskleeh 

I was able to recreate the problem with `mn create-app appname`.

After generating the app, `./gradlew sJ` demonstrates the issue...

```
~ $ mn create-app issue487
| Application created at /Users/jeffbrown/issue487
~ $ cd issue487/
issue487 $ ./gradlew sJ

Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
See https://docs.gradle.org/4.8/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 4s
3 actionable tasks: 3 executed
```

**Comment by jameskleeh** [comment]
Here is the issue on the shadow plugin tracker https://github.com/johnrengelman/shadow/issues/336

**Comment by jameskleeh** [comment]
Once they provide a solution, we can update the CLI to implement it

**Comment by jeffbrown** [comment]
FYI... I have confirmed that if `mainClassName = "somepackage.Application"` is after `apply plugin:"application"` and before `apply plugin:"com.github.johnrengelman.shadow"` that the warning doesn't appear to be logged.

**Comment by aplatypus** [comment]
Will / would the annotation be ignored?
-  "Detecting annotation processors on the compile classpath is deprecated and Gradle 5.0 will ignore them."

I have this message on Java and Groovy files under a `groovy/` directory.  Files under a `java/` tree seem to be OK.  I'm not knowingly using the `shadow plugin`.  Is that something included internally?

**Comment by jameskleeh** [comment]
@aplatypus That issue doesn't have anything to do with the shadow plugin. Do you have `annotationProcessor "io.micronaut:inject-java"` in your gradle dependencies?

**Comment by jameskleeh** [comment]
The issue has been fixed upstream, just waiting on a release

**Comment by aplatypus** [comment]
@jameskleeh ... yes ido.

**Comment by jameskleeh** [comment]
@aplatypus like I said that is a separate issue. Feel free to create a new issue with an example project
